Userland/Applications/GLClockDemo/GLClock.h

```diff
@@ -0,0 +1,234 @@
+            float outerRadius = 1.0f;
+
+            float x1 = innerRadius * cos(angle);
+            float y1 = innerRadius * sin(angle);
+            float x2 = outerRadius * cos(angle);
+            float y2 = outerRadius * sin(angle);
+
+            draw_line(x1, y1, x2, y2, 0.025f, Color(255, 0, 0));
+        }
+    }
+    static void minute_markings()
+    {
+        for (int i = 0; i < 60; ++i) {
+            if (i % 5 != 0) {
+                float angle = (2.0 * M_PI / 60) * i;
+                float innerRadius = 0.95f; // 95% of the clock's radius
+                float outerRadius = 1.0f;
+
+                float x1 = innerRadius * cos(angle);
+                float y1 = innerRadius * sin(angle);
+                float x2 = outerRadius * cos(angle);
+                float y2 = outerRadius * sin(angle);
+
+                draw_line(x1, y1, x2, y2, 0.01f, Color(0, 0, 200));
+            }
+        }
+    }
+
+    virtual void timer_event(Core::TimerEvent&) override
+    {
+        update();
+        set_time();
+    }
+
+public:
+    String timezone() const { return m_timezone; }
+
+    void set_timezone(String tooltip)
+    {
+        m_timezone = move(tooltip);
+    }
+
+    virtual ~GLClock() override = default;
+
+protected:
+    void initializeGL() override
+    {
+        set_time();
+        glClearColor(0, 0, 0, 0);
+    }
+
+    void paintGL() override
+    {
+        glClear(GL_COLOR_BUFFER_BIT);
+
+        glPushMatrix();
+        glRotatef(90, 0.0, 0.0, 1.0);
+
+        draw_circle(1.0f, 100, Color(255, 255, 255));
+
+        hour_markings();
+        minute_markings();
+
+        auto seconds = m_time.second();
+        auto minutes = m_time.minute();
+        auto hours = m_time.hour() % 12;
+
+        float secondsAngle = -(2.0 * M_PI / 60) * seconds;
+
+        float minuteAngle = -(2.0 * M_PI / 60) * minutes;
+
+        float hourAngle = -(2.0 * M_PI / 12) * (hours + static_cast<double>(minutes) / 60.0);
+
+        draw_hand(0.7f, hourAngle, Color(0, 0, 255));
+        draw_hand(0.8f, minuteAngle, Color(0, 255, 0));
+        draw_hand(0.8f, secondsAngle, Color(255, 0, 0));
+        draw_circle(0.05f, 50, Color(0, 0, 0, 255));
+        glPopMatrix();
+    }
+
+    void resizeGL(int w, int h) override
+    {
+        glViewport(0, 0, w, h);
+
+        glMatrixMode(GL_PROJECTION);
+        glLoadIdentity();
+
+        double aspect_ratio = static_cast<double>(w) / static_cast<double>(h);
+        if (aspect_ratio > 1.0) {
+            glOrtho(-2.0 * aspect_ratio, 2.0 * aspect_ratio, -2.0, 2.0, -1, 1);
+        } else {
+            glOrtho(-2.0, 2.0, -2.0 / aspect_ratio, 2.0 / aspect_ratio, -1, 1);
+        }
+
+        glMatrixMode(GL_MODELVIEW);
+        glLoadIdentity();
+        if (aspect_ratio > 1.0) {
+            glScalef(1.0 / aspect_ratio, 1.0, 1.0); // Scale down X dimension
+        } else {
+            glScalef(1.0, aspect_ratio, 1.0); // Scale down Y dimension
+        }
+    }
```

Comment:
Why have default implementations for initialize / paint / resize if you're going to overwrite them in the subclass?

Reply:
The aim is to show a black surface by deafult

Reply:
But a black surface is the default output of a GLContext without doing anything at all as well, right?